### ISSUE TITLE
S2187: Rule should not raise False Positives for MSTEST Base class with only TestCleanup and TestInitialize methods

### ISSUE THREAD

**Issue by Shakevg** [SonarSource/sonar-dotnet#5394]
### Description

We are getting [S2187](https://sonarsource.github.io/rspec/#/rspec/S2187/csharp) warnings for MSTEST base test classes that have no test directly but have TestCleanup and TestInitialize methods.

### Repro steps

Our tests usually follow the class inheritance of the BaseTests class with a common method for tests: cleanup and test init.

```csharp
    [TestClass]
    public class TestDemo : BaseTests
    {
        [TestMethod]
        public void SaveOrder_OnlyRequiredValuesFilled_SuccessfullySaved()
        {
         
        }

    }

    [TestClass]
    public class BaseTests
    {
        [TestInitialize]
        public void Setup()
        {
            Console.WriteLine("Setup executed.");
           
        }

        [TestCleanup]
        public void Cleanup()
        {
            Console.WriteLine("Cleanup executed.");
           
        }
    }
```

### Expected behavior

No warning.

### Actual behavior

Warning that BaseTests class has no tests.

### Known workarounds

Suppress warning

### Related information

SonarC# Version 8.34
Visual Studio Version 17.0

**Comment by pavel-mikula-sonarsource** [comment]
Hi @Shakevg,

Your base class should be abstract and it shouldn't contain `TestClassAttribute`, because it doesn't contain any `TestMethodAttribute`. After fixing that, the S2187 doesn't raise as expected.

See MS docs: https://docs.microsoft.com/en-us/previous-versions/ms182517(v=vs.110)
